src/controllers/post.js

```diff
@@ -11,18 +12,34 @@ export const create = async (req, res) => {
 }
 
 export const getAll = async (req, res) => {
-  return sendDataResponse(res, 200, {
-    posts: [
-      {
-        id: 1,
-        content: 'Hello world!',
-        author: { ...req.user, firstName: 'Test', lastName: 'McTesterson' }
-      },
-      {
-        id: 2,
-        content: 'Hello from the void!',
-        author: { ...req.user, firstName: 'Bobby', lastName: 'McTesterson' }
+  const getPosts = await getAllPostsDb()
+  const posts = getPosts.map((post) => {
+    const { user } = post
+    if (user && user.profile) {
```

Comment:
Let's reverse this if statement. Instead check if we don't have user or use profile, then return post.

Reply:
Happy to make those changes to check if the profile exists on the user. We agreed that it doesn't make sense to check if the user exists in the if statement on the post as a user must be logged-in, in order to create a post so this will always be true.